File: sdks/ruby/lib/novabilling/types/mrr_plan_breakdown.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class MrrPlanBreakdown < Internal::Types::Model
      field :plan_id, -> { String }, optional: false, nullable: false, api_name: "planId"
      field :plan_name, -> { String }, optional: false, nullable: false, api_name: "planName"
      field :mrr, -> { Integer }, optional: false, nullable: false
      field :subscription_count, -> { Integer }, optional: false, nullable: false, api_name: "subscriptionCount"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/MrrPlanBreakdown.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record MrrPlanBreakdown : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("planId")]
    public required string PlanId { get; set; }

    [JsonPropertyName("planName")]
    public required string PlanName { get; set; }

    [JsonPropertyName("mrr")]
    public required double Mrr { get; set; }

    [JsonPropertyName("subscriptionCount")]
    public required double SubscriptionCount { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
